source/base/parameter_handler.cc

```diff
@@ -1708,33 +1699,31 @@ bool ParameterHandler::read_input (const std::string &filename,
     {
       std::string openname = search.find(filename);
       std::ifstream file_stream (openname.c_str());
-      AssertThrow(file_stream, ExcIO());
-
-      return read_input (file_stream, filename, last_line);
+      read_input (file_stream, filename, last_line);
     }
   catch (const PathSearch::ExcFileNotFound &)
     {
-      std::cerr << "ParameterHandler::read_input: could not open file <"
-                << filename << "> for reading." << std::endl;
+      // Note that, when ParameterHandler's methods signaled failure with
+      // return codes, catching the exception here as part of control flow was
+      // a reasonable way to use this function (indeed, it was a common way to
```

Comment:
to understand this comment requires knowing the history of the file. that's not going to be useful ten years from now -- I would just describe what the function does right now.